packages/discovery-react-components/src/components/DocumentPreview/components/HtmlView/HtmlView.tsx

```diff
@@ -52,9 +52,11 @@ export const HtmlView: FC<Props> = ({
   const contentRef = useRef<HTMLDivElement>(null);
   const highlightRef = useRef<HTMLDivElement>(null);
 
-  if (setHideToolbarControls) {
-    setHideToolbarControls(true);
-  }
+  useEffect(() => {
+    if (setHideToolbarControls) {
+      setHideToolbarControls(true);
+    }
+  }, [setHideToolbarControls]);
```

Comment:
Oh, is this all it took? I thought we were going to have to use the `useEffect` cleanup function. Good catch.